MAINTAINERS.yml

```diff
@@ -0,0 +1,1025 @@
+    status: maintained
+    maintainers:
+        - rveerama1
+    files:
+        - subsys/net/lib/coap/
+        - samples/net/sockets/coap_*/
+        - tests/net/lib/coap/
+
+"Networking: LWM2M":
+    # Provisional entry, to be corrected
+    status: maintained
+    maintainers:
+        - mike-scott
+    files:
+        - samples/net/lwm2m_client/
+        - subsys/net/lib/lwm2m/
+    labels:
+        - "area: LWM2M"
+
+"Networking: MQTT":
+    # Provisional entry, to be corrected
+    maintainers:
+        - rlubos
+    collaborators:
+        - jukkar
+    files:
+        - subsys/net/lib/mqtt/
+        - tests/net/lib/mqtt_packet/
+        - samples/net/mqtt_publisher/
+
+"NIOS-2 arch":
+    # Provisional entry, to be corrected
+    # TODO: The entries in the maintainers spreadsheet do not match CODEWONERS at all!
+    status: maintained
+    maintainers:
+        - ruuddw
+    collaborators:
+        - vonhust
+    files:
+        - arch/nios2/
+        - include/arch/nios2/
+    labels:
+        - "area: NIOS2"
+
+"nRF52 BSIM":
+    status: maintained
+    maintainers:
+        - aescolar
+    files:
+        - boards/posix/nrf52_bsim/
+    labels:
+        - "platform: nrf52_bsim"
+
+"POSIX API layer":
+    # Provisional entry, to be corrected
+    status: maintained
+    maintainers:
+        - pfalcon
+    files:
+        - include/posix/
+        - lib/posix/
+        - tests/posix/
+    labels:
+        - "area: POSIX"
+
+"Power management":
+    # TSC : Note this entry differs from the spreadsheet:
+    # pizi-nordic is not anymore working in Zephyr, therefore he cannot be a
+    # maintainer, and the are is orphaned until a new maintainer is appointed
+    status: orphaned
+    collaborators:
+        - wentongwu
+    files:
+        - include/power/power.h
+        - samples/subsys/power/
+        - subsys/power/
+    labels:
+        - "area: Power Management"
+
+"RISCV arch":
+    # Provisional entry, to be corrected
+    status: maintained
+    maintainers:
+        - nategraff-sifive
+    collaborators:
+        - mgielda
+    files:
+        - arch/riscv/
+        - include/arch/riscv/
+    labels:
+        - "area: RISCV"
+
+Sanitycheck:
+    # Provisional entry, to be corrected
+    status: maintained
+    maintainers:
+        - nashif
+    files:
+        - scripts/sanitycheck
+        - scripts/sanity_chk/
+    labels:
+        - "area: Sanitycheck"
+
+Settings:
+    # Provisional entry, to be corrected
+    status: maintained
+    maintainers:
+        - nvlsianpu
+    files:
+        - include/settings/
+        - subsys/settings/
+        - tests/subsys/settings/
+    labels:
+        - "area: Settings"
+
+Shell:
+    # Provisional entry, to be corrected
+    status: maintained
+    maintainers:
+        - jakub-uC
+    files:
+        - include/shell/
+        - samples/subsys/shell/
+        - subsys/shell/
+        - tests/shell/
+        - tests/subsys/shell/
+    labels:
+        - "area: Shell"
+
```

Comment:
"STM32":
    status: maintained
    maintainers:
        - erwango
    files:
        - boards/arm/nucleo_*
        - boards/arm/stm32*_disco
        - boards/arm/stm32*_eval
        - drivers/*/*stm32*
        - drivers/*/*/*stm32*
        - dts/arm/st
        - dts/bindings/*/*stm32*
        - soc/arm/st_stm32
    labels:
        - "platform: STM32"
    description: >-
        STM32 SOCs, dts files and related drivers. ST nucleo, disco and eval boards.

Reply:
This addition would cause lots of drivers to have more than one maintainer.

Reply:
Thank you very much @erwango. Having some components being covered by more than 1 maintainer in different dimensions is, in principle, not a problem (The maintainers would mostly just need to be civil with each other. If they can agree about the direction those common files should go everything should be fine. If they can't, matters can be escalated to the TSC, though hopefully that should not be the case too often).
But I would not encourage having too many maintainers of the same thing to avoid ending too often in deadlocks and needing to escalate to the TSC.
In any case, following process, this entry in itself would need also TSC approval (as unfortunately there is nothing equivalent in the spreadsheet), so I add it to the list.
Loud thinking: It seems I may need to keep some of this "for TSC approval" patches as separate commits.

Reply:
> Having some components being covered by more than 1 maintainer in different dimensions is, in principle, not a problem.

This is indeed the current way of working for current STM32 drivers, where I am ensuring that reviews of all STM32 drivers are done either by the driver codeowner (and in that case I may provide a co-review but let the final approval to the codeowner) either by known driver users and myself. This has been the case for several years now and I don't think there are much complaints on this way of working.

Reply:
>and in that case I may provide a co-review but let the final approval to the codeowner

For being informed and providing feedback, a collaborator would be enough right?

Generally, I have no objections adding @erwango as the STM32  maintainer. I just want to mention that conflicts may arise.

Do we add maintainers or collaborators for the other SoCs too?

Reply:
> Do we add maintainers or collaborators for the other SoCs too?

I think we should. But we don't have entries yet.

Reply:
> For being informed and providing feedback, a collaborator would be enough right?

This is not the only point, it also consists in ensuring consistency in whole STM32 drivers, dependencies with STM32Cube (when applying) or other drivers (like clock_control today, and potentially pinmux tomorrow, power management, ...).
Last, it may also consists in identifying/promoting/acknowledging maintainers on new drivers or identifying drivers that are no longer maintained (unfortunate, but in the long range, this kind of things happen) and trigger consecutive actions.

Reply:
@erwango I totally agree with you and appreciate having someone doing that work. My concerned are regarding the script that will assign the assignee in the future. The assignee should be the subsys maintainer. Maybe a platform maintainer role could help solving this.
@aescolar what do you think? Should this be discussed in the next process meeting? It would really be nice if we have maintainers for all platforms.